### PR TITLE
Feature/md5 registration store

### PR DESCRIPTION
**Pull request by BabysbreathJJ**
Add md5 check when download & move config file.

### CODE REVIEW

File: src/TelepathyCommon.UnitTest/AzureBlobServiceRegistrationStoreUnitTest.cs [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        private const string storageConnectionString = "UseDevelopmentStorage=true;";
+
+        private CloudStorageAccount storageAccount;
+
+        private AzureBlobServiceRegistrationStore serviceRegistrationStore = new AzureBlobServiceRegistrationStore(storageConnectionString);
+
+        [TestMethod]
+        public async Task CalculateMd5HashEmptyTest()
+        {
+            if (CloudStorageAccount.TryParse(storageConnectionString, out storageAccount))
```
Comment (amat27) [inline src/TelepathyCommon.UnitTest/AzureBlobServiceRegistrationStoreUnitTest.cs]: When this `if` returns `false`, this case succeeds or fails?

File: src/TelepathyCommon.UnitTest/AzureBlobServiceRegistrationStoreUnitTest.cs [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+                string localPath = Environment.CurrentDirectory + "\\blobs\\";
+                string localFileName = "ccpechosvc_empty.config";
+                string sourceFile = Path.Combine(localPath, localFileName);
+
+                Console.WriteLine("Temp file = {0}", sourceFile);
+                Console.WriteLine("Uploading to Blob storage as blob '{0}'", localFileName);
+
+                CloudBlockBlob cloudBlockBlob = cloudBlobContainer.GetBlockBlobReference(localFileName);
+                await cloudBlockBlob.UploadFromFileAsync(sourceFile);
```
Comment (amat27) [inline src/TelepathyCommon.UnitTest/AzureBlobServiceRegistrationStoreUnitTest.cs]: is `UploadFromStream` and use a memory stream better?

File: src/TelepathyCommon.UnitTest/AzureBlobServiceRegistrationStoreUnitTest.cs [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+            Console.WriteLine("Init md5 in storage : {0}", initMd5);
+            Console.WriteLine("Init calculated md5 : {0}", initCalculatedMd5);
+            Assert.AreEqual(initMd5, initCalculatedMd5);
+
+            await serviceRegistrationStore.SetAsync("testservice", new Version("1.0"), "Test service version 1.0, content changed");
+            filePath = await serviceRegistrationStore.ExportToTempFileAsync("testservice", new Version("1.0"));
+            string fileContent = File.ReadAllText(filePath);
+            Console.WriteLine("File path : {0}", filePath);
+            Console.WriteLine("Changed file content : {0}", fileContent);
+            Assert.AreEqual("Test service version 1.0, content changed", fileContent);
```
Comment (amat27) [inline src/TelepathyCommon.UnitTest/AzureBlobServiceRegistrationStoreUnitTest.cs]: Save the expected value in to a constant

File: src/TelepathyCommon/ServiceRegistrationStore/AzureBlobServiceRegistrationStore.cs [hunk 4]
```diff
@@ -43,17 +44,29 @@ public async Task<List<string>> EnumerateAsync()
             return (await this.blobContainer.ListBlobsSegmentedAsync(null, null)).Results.Select(b => Path.GetFileNameWithoutExtension(b.Uri.ToString())).ToList();
         }
 
-        // TODO: Design - do we still need check MD5 here?
+        public string CalculateMd5Hash(byte[] blobData)
+        {
+            var md5 = MD5.Create();
```
Comment (amat27) [inline src/TelepathyCommon/ServiceRegistrationStore/AzureBlobServiceRegistrationStore.cs]: `MD5` is `IDisposable`.  Either dispose it after use or save it.

File: src/TelepathyCommon.UnitTest/AzureBlobServiceRegistrationStoreUnitTest.cs (hunk 1 above)
Comment (BabysbreathJJ) [reply src/TelepathyCommon.UnitTest/AzureBlobServiceRegistrationStoreUnitTest.cs]: Fails. Error message: "No valid combination of account information found."

File: src/TelepathyCommon.UnitTest/AzureBlobServiceRegistrationStoreUnitTest.cs (hunk 2 above)
Comment (BabysbreathJJ) [reply src/TelepathyCommon.UnitTest/AzureBlobServiceRegistrationStoreUnitTest.cs]: Have changed to use `UploadTextAsyn.`

**Pull request merged**